### ISSUE TITLE
Issues are not shown in the Changelog.

### ISSUE THREAD

**Issue by ortegaantunes** [nemerosa/ontrack#984]
Hi @dcoraboeuf 

We are facing some issues when we try to get the Issues from the Changelog page.

Ontrack version: 4.2.7

This is the behaviour when we try to hit the "Issues" feature.

![Screenshot 2022-05-27 at 00 08 01](https://user-images.githubusercontent.com/29414973/170589145-7e7c48ed-365b-42a0-92c3-07eec6e21a71.png)

On Ontrack, we got these logs:

<img width="1115" alt="Screenshot 2022-05-27 at 00 09 22" src="https://user-images.githubusercontent.com/29414973/170588035-3a9d96fa-6caf-47cd-b253-f9d4873f6abe.png">

<img width="1276" alt="Screenshot 2022-05-27 at 00 09 32" src="https://user-images.githubusercontent.com/29414973/170588071-be2a196f-ae07-4546-b4b3-a6f45dcac851.png">

![Screenshot 2022-05-27 at 00 24 28](https://user-images.githubusercontent.com/29414973/170589515-eaaca54c-e8a4-45ff-bb10-0df0489dec17.png)

In the above images, we can see a message as "No issue service has been configured", indeed at the "Combined issue services configurations" page no configuration is shown.

![Screenshot 2022-05-26 at 23 56 38](https://user-images.githubusercontent.com/29414973/170588216-306ccdc9-d7b7-40d7-9b69-48aa490ab939.png)

But we already have this configuration and it was working until the last update(4.0.54 to 4.2.7). Performing a select into the database we got this value.
(SELECT id, type, name, content FROM public.configurations;)

![Screenshot 2022-05-27 at 00 04 27](https://user-images.githubusercontent.com/29414973/170588281-204309ab-a011-42aa-bc51-e3b75710f488.png)

As we have talked about, this is the current scenario.

Any suggestion?

Thanks in advance

**Comment by dcoraboeuf** [comment]
The following issues:

* (1) malicious `//`
* (3) combined configurations page access

are fixed (not released yet).

For (2) No issue service has been configured, I still need to reproduce the issue.

My question is still valid though: the combined issue service is used only in very specific cases, when aggregating issues from different issue services (GitHub issues & Jira for example). It seems that in your case you only use Jira, so it might be better to use a direct Jira configuration?

**Comment by dcoraboeuf** [comment]
I've created a project using a combined issue service pointing to an existing Jira configuration, and I've no issue getting a change log.

Can you provide me with the result of the following query ?

```sql
select pp.json
from properties pp
inner join projects p on p.id = pp.project
where p.name = '<your project name>'
and pp.type = 'net.nemerosa.ontrack.extension.github.property.GitHubProjectConfigurationPropertyType'
```

**Comment by ortegaantunes** [comment]
Hi Damien

To be honest I don't recall why we have defined it in this way instead  of Jira direct. We also have the Jira integration configured. How can we define it into to the project via ontrack-jenkins-cli-pipeline? There isn't any reason for us to use the combined issue feature. So, we can switch to Jira property.

Thanks

Obtener Outlook para Android<https://aka.ms/AAb9ysg>

________________________________
De: Damien Coraboeuf ***@***.***>
Enviado: viernes, 27 de mayo de 2022 12:06
Para: nemerosa/ontrack ***@***.***>
CC: Rafael Ortega Antunes ***@***.***>; Mention ***@***.***>
Asunto: Re: [nemerosa/ontrack] Issues are not shown in the Changelog. (Issue #984)


The following issues:

  *   (1) malicious //
  *   (3) combined configurations page access

are fixed (not released yet).

For (2) No issue service has been configured, I still need to reproduce the issue.

My question is still valid though: the combined issue service is used only in very specific cases, when aggregating issues from different issue services (GitHub issues & Jira for example). It seems that in your case you only use Jira, so it might be better to use a direct Jira configuration?

**Comment by dcoraboeuf** [comment]
If you're using [`ontrackCliSetup`](https://github.com/nemerosa/ontrack-jenkins-cli-pipeline/blob/main/vars/ontrackCliSetup.md#ontrackclisetup) you can use the `scmIssues` property:

```groovy
ontrackCliSetup(
  // ...
  scmIssues: "jira//Jira" // <-- jira// followed by the exact name of your Jira configuration
)
```

Note that if you have many repositories using the same configuration, you can define a global `ONTRACK_SCM_ISSUES` Jenkins environment variable instead.

**Comment by dcoraboeuf** [comment]
Then, I really need the result of the SQL query above to look a bit more into your configuration :D

**Comment by ortegaantunes** [comment]
@dcoraboeuf the query does not return any values.

**Comment by dcoraboeuf** [comment]
If you're using Bitbucket, the query must be adapted:

* for Bitbucket Server:

```sql
select pp.json
from properties pp
inner join projects p on p.id = pp.project
where p.name = '<your project name>'
and pp.type = 'net.nemerosa.ontrack.extension.stash.property.StashProjectConfigurationPropertyType'
```

* for Bitbucket Cloud:

```sql
select pp.json
from properties pp
inner join projects p on p.id = pp.project
where p.name = '<your project name>'
and pp.type = 'net.nemerosa.ontrack.extension.bitbucket.cloud.property.BitbucketCloudProjectConfigurationPropertyType'
```

**Comment by ortegaantunes** [comment]
![Screenshot 2022-05-27 at 15 14 18](https://user-images.githubusercontent.com/29414973/170706465-26928820-670e-45cd-8134-61353189ff66.png)

This is the result.

**Comment by dcoraboeuf** [comment]
`Jira-Issues` as an identifier is not correct. It should look like `jira//<name>` where `<name>` is the name of your Jira configuration in Ontrack (I believe it's `Jira` in your case?).

**Comment by ortegaantunes** [comment]
![Screenshot 2022-05-27 at 15 19 38](https://user-images.githubusercontent.com/29414973/170707238-75e69b26-dce4-4458-8577-2a3dce18cd60.png)
We have both names, just pointing to the same Jira organization(only test purpose). But, What I've understood is that we need to have this configuration like that -> scmIssues: "jira//Jira" or scmIssues: "jira//Jira-Issues", right?

**Comment by dcoraboeuf** [comment]
Indeed. Either `jira//Jira` or `jira//Jira-Issues`.

**Comment by ortegaantunes** [comment]
Ok. Is there any way to update it via Database?

**Comment by dcoraboeuf** [comment]
Normally, I always prefer define my configurations as code, like in a Jenkinsfile, using `ontrackCliSetup` for example and `scmIssues`. But you should be also able to use the UI: go to your project page. In the list of properties, you should see something like "Bitbucket configuration" and you have a pencil icon to edit it.

What I would be interested into is how you got this in the first place, because maybe some controls are missing.

**Comment by ortegaantunes** [comment]
Yes, I agree with you, actually, we already have it via Jenkins Shared Library, I'll update that. The question was just to avoid waiting until the next Project interaction, but no worries. It is already working as expected. Regarding how it was working I don't know to be honest hehe. Maybe before the Bitbucket cloud was supported, If I'm not mistaken before that we were using Git Integration to connect to Bitbucket Cloud.

What I can say is we never had the configuration as scmIssues: "jira//Jira"

One year ago for example(I'll buy a cake to celebrate =))
![Screenshot 2022-05-27 at 15 40 55](https://user-images.githubusercontent.com/29414973/170711497-5e62f6fc-5436-4393-b93d-af3ba54d79c3.png)

**Comment by dcoraboeuf** [comment]
Ok, thanks for the feedback all the same. I'll wait for your final tests, and when OK, I'll still release the two small changes I've done. This cannot hurt.

**Comment by ortegaantunes** [comment]
All good @dcoraboeuf. I've changed some Projects manually, and so far, everything is working.

**Comment by dcoraboeuf** [comment]
Good to know 👌 I'll do a quick release for the two small fixes (the rogue call & the list of combined issue configurations)

**Comment by dcoraboeuf** [comment]
Fixes available in 4.2.14

**Comment by ortegaantunes** [comment]
Sure, I'm away from my laptop atm. I will send to you asap. =)

Obtener Outlook para Android<https://aka.ms/AAb9ysg>
________________________________
